components/ui/location-combobox.tsx

```diff
@@ -0,0 +1,185 @@
+import {
+  Popover,
+  PopoverContent,
+  PopoverTrigger,
+} from "@/components/ui/popover";
+
+type Location = {
+  name: string;
+  type: string;
+  regionId: string;
+  coordinates: [number, number];
+  country: string;
+  domain: string;
+};
+
+// TODO: Replace with a DB call to get the cached regions
+const cachedLocations: Location[] = [
+  {
+    name: "San Jose, California, United States",
+    type: "CITY",
+    regionId: "602703",
+    coordinates: [37.3354, -121.891907],
+    country: "United States",
+    domain: "US",
+  },
+
+  {
+    name: "San Jose (SJC - Norman Y. Mineta San Jose Intl.), California, United States",
+    type: "AIRPORT",
+    regionId: "4409939",
+    coordinates: [37.369739, -121.929225],
+    country: "United States",
+    domain: "US",
+  },
+
+  {
+    name: "San José, San José Province, Costa Rica",
+    type: "CITY",
+    regionId: "3177",
+    coordinates: [9.93286, -84.079559],
+    country: "Costa Rica",
+    domain: "CR",
+  },
+
+  {
+    name: "San José del Cabo, Baja California Sur, Mexico",
+    type: "CITY",
+    regionId: "8650",
+    coordinates: [23.063656, -109.702438],
+    country: "Mexico",
+    domain: "MX",
+  },
+
+  {
+    name: "San José (SJO - Juan Santamaría Intl.), Costa Rica",
+    type: "AIRPORT",
+    regionId: "5196371",
+    coordinates: [9.9981, -84.2046],
+    country: "Costa Rica",
+    domain: "CR",
+  },
+
+  {
+    name: "San Jose Convention Center, San Jose, California, United States",
+    type: "POI",
+    regionId: "6138778",
+    coordinates: [37.329081, -121.889132],
+    country: "United States",
+    domain: "US",
+  },
+
+  {
+    name: 'San José del Cabo ("San Jose Del Cabo Airport") (SJD - Los Cabos Intl.), Baja California Sur, Mexico',
+    type: "AIRPORT",
+    regionId: "4475475",
+    coordinates: [23.162746, -109.717368],
+    country: "Mexico",
+    domain: "MX",
+  },
+
+  {
+    name: "Downtown San Jose, San Jose, California, United States",
+    type: "NEIGHBORHOOD",
+    regionId: "553248633938970011",
+    coordinates: [37.335095, -121.892883],
+    country: "United States",
+    domain: "US",
+  },
+
+  {
+    name: "San Jose (SJO - All Airports), Costa Rica",
+    type: "METROCODE",
+    regionId: "6139076",
+    coordinates: [9.93960791512362, -84.10325688819586],
+    country: "Costa Rica",
+    domain: "CR",
+  },
+
+  {
+    name: "San Jose State University, San Jose, California, United States",
+    type: "POI",
+    regionId: "6069527",
+    coordinates: [37.32890082806852, -121.8749250818403],
+    country: "United States",
+    domain: "US",
+  },
+];
+
+type LocationComboboxProps = {
+  searchLocations: (query: string) => void;
+  setRegionId: (regionId: string) => void;
+};
+
+export default function LocationCombobox(props: LocationComboboxProps) {
+  const [open, setOpen] = React.useState(false);
+  const [value, setValue] = React.useState("");
+  const handleSearch = () => {
+    setValue("");
+    props.searchLocations(value);
+  };
```

Comment:
```suggestion
  const handleSearch = (value: string) => {
      setValue(value);
      props.searchLocations(value);
  };
```

The input for the API `searchLocations` is not being updated properly.

Reply:
Also, add another useState hook to store the region id that the user selected from the API or cached locations.